app/Auth0/Auth0TenantSDK.php

```diff
@@ -0,0 +1,95 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Auth0;
+
+use App\Domain\Integrations\Integration;
+use App\Domain\Integrations\IntegrationType;
+use App\Json;
+use Auth0\SDK\Auth0;
+use Auth0\SDK\Configuration\SdkConfiguration;
+use Auth0\SDK\Contract\API\ManagementInterface;
+use Psr\Http\Message\ResponseInterface;
+
+final class Auth0TenantSDK
+{
+    private ManagementInterface $management;
+
+    public function __construct(public readonly Auth0Tenant $auth0Tenant, SdkConfiguration $sdkConfiguration)
+    {
+        $auth0 = new Auth0($sdkConfiguration);
+
+        if (!$auth0->configuration()->hasManagementToken()) {
+            $response = $auth0->authentication()->clientCredentials($sdkConfiguration->getAudience());
+            $response = Json::decodeAssociatively((string) $response->getBody());
+            $auth0->configuration()->setManagementToken($response['access_token']);
+        }
+
+        $this->management = $auth0->management();
+    }
+
+    public function createClientForIntegration(Integration $integration): Auth0Client
+    {
+        $name = sprintf('%s (id: %s)', $integration->name, $integration->id->toString());
+
+        $apis = match ($integration->type) {
+            IntegrationType::SearchApi, IntegrationType::Widgets => 'sapi',
+            IntegrationType::EntryApi => 'entry sapi',
+        };
```

Comment:
Cool php has match expressions 🙂

Reply:
@simon-debruijn yes but it's specifically for [matching possible values of enums](https://www.php.net/manual/en/control-structures.match.php). You are maybe comparing it to rust match expressions? In that case https://www.php.net/manual/en/control-structures.switch.php is more comparable I think